### ISSUE TITLE
oc: rolling back to the same dc version should be a no-op

### ISSUE THREAD

**Issue by kargakis** [openshift/origin#9113]
Not currently:

```
[vagrant@localhost sample-app]$ oc get dc
NAME       REVISION   REPLICAS   TRIGGERED BY
database   2          1          config
frontend   0          2          config,image(origin-ruby-sample:latest)
[vagrant@localhost sample-app]$ oc rollback dc/database --to-version=2
#3 rolled back to database-2
[vagrant@localhost sample-app]$ oc get dc
NAME       REVISION   REPLICAS   TRIGGERED BY
database   3          1          config
frontend   0          2          config,image(origin-ruby-sample:latest)
```

**Comment by johnplanow** [comment]
Hey @kargakis , would you mind taking a look at what I've done and let me know if I've headed in the right direction? This is my first work on the OpenShift project, so please don't hesitate to tell me if I haven't gone about it correctly. I'll try to get online during your workday in case we need to discuss. Thanks!

Here's the change:

https://github.com/johnplanow/origin/commit/3a71fd6cdade88f8e5145169379813be07f8688f

And here's the current output:

`stack@devstack1: sample-app$ oc get dc
NAME       REVISION   DESIRED   CURRENT   TRIGGERED BY
database   1          1         1         config
frontend   1          2         2         config,image(origin-ruby-sample:latest)
stack@devstack1: sample-app$ oc rollback dc/frontend --to-version=1
error: version 1 is already the latest version
stack@devstack1: sample-app$ oc get dc
NAME       REVISION   DESIRED   CURRENT   TRIGGERED BY
database   1          1         1         config
frontend   1          2         2         config,image(origin-ruby-sample:latest)
`

**Comment by kargakis** [comment]
Hi @johnplanow 

the check looks good but it should be in the [rollback rest](https://github.com/openshift/origin/blob/42bfa0d5c5e1edf63d7a3df271840e2c0390db99/pkg/deploy/registry/rollback/rest.go#L72) and not part of the cli code in order for it to work also for `oc rollout undo`.

**Comment by matthyx** [comment]
Hey @kargakis @johnplanow , what's the current status of this issue? Should I have a look?

**Comment by kargakis** [comment]
If @johnplanow is not working on this anymore, then I guess you can pick it up.

**Comment by johnplanow** [comment]
My apologies -- I spent about a half a day on this before starting a new job and haven't been able to get back to it. I should have left a comment here at least. 

@matthyx , please have a look if you'd like.

**Comment by matthyx** [comment]
Ok, you can assign it to me then. Cheers.

**Comment by kargakis** [comment]
@matthyx issues can be assigned only to members of the openshift organization but feel free to start working on it.

**Comment by matthyx** [comment]
Ok, consider me assigned to it... we'll sort the membership issue later ;-)

**Comment by matthyx** [comment]
The fix is quite minimal in fact: https://github.com/openshift/origin/pull/13104

**Comment by matthyx** [comment]
I think this issue can be closed, or marked as solved for next release?

**Comment by pweil-** [comment]
Fixed with https://github.com/openshift/origin/pull/13104.  closing
